app.js

```diff
@@ -186,6 +199,50 @@ async function setupLookup() {
   }
 }
 
+function verify(req) {
+  if (!req.get('X-Signature-Ed25519') || !req.get('X-Signature-Timestamp')) {
+    return false;
+  }
+
+  const signature = req.get('X-Signature-Ed25519');
+  const timestamp = req.get('X-Signature-Timestamp');
+  const body = req.rawBody; // rawBody is expected to be a string, not raw bytes
+
+  return nacl.sign.detached.verify(
+    Buffer.from(timestamp + body),
+    Buffer.from(signature, 'hex'),
+    Buffer.from(PUBLIC_KEY, 'hex')
+  );
+}
+
+app.post('/discord', async (req, res) => {
+  if (!verify(req)) {
+    res.status(401).send();
+    return;
+  }
+  const interaction = req.body;
+  if (interaction.type == 1) { // ping
+    res.json({'type': 1}); // pong
+  }
+  if (interaction.type == 2) { // APPLICATION_COMMAND
+    const data = interaction.data;
+    if (data.name == "dldialogue") {
+      const name = data.options[0].value;
+      const message = data.options[1].value;
+      const nameEnc = encodeURI(name);
+      const messageEnc = encodeURI(message);
+      res.json({
+        'type': 4,
+        'data': {
+          'content': "http://api.dldialogue.xyz/" + nameEnc + "/" + messageEnc
+        }
+      });
+    }
+  }
+  res.status(404).send();
```

Comment:
This looks like it'll cause an error in the console. Recall that `res.send` does not terminate the function, so the code will continue on to this line, which will attempt to set the status of an already sent response. This will result in an error in the console.

Prefer putting this in an else block, or alternatively using return to escape early from the function (not 154-style). This is similar to the change above regarding if => else if.

Reply:
Diving deeper into this -- calling `res.status` does not actually set a header in express code, so the code that throws this error does not execute. This will result in an error though if any other header is set (i.e. a call to `res.type`). It also just isn't great to overwrite and send again as it is a little confusing. I believe the logic flow makes more sense if this case is it's own `else` block away from the happy path.